### PR TITLE
642 handle reprojection of sis not in the same projection as the hydraulics

### PR DESCRIPTION
**Pull request by Brennan1994**

### CODE REVIEW

**Comment by Brennan1994** [comment]
! I'm seeing the memory issue again on this branch. I'm not sure if it's present on Main, or if I did something to trigger it. 

Needs investigatoin.

**Comment by Brennan1994** [comment]
I'm using a impact area set with 3 areas and a muncie SI. Not sure why this is throwing an issue, but it's happening on main too. Will bring up tomorrow.

**Review by rnugent3 — APPROVED** [review]
awesome, thanks, @Brennan1994

File: fda-model/structures/Inventory.cs [hunk 1]
```diff
@@ -158,112 +111,115 @@ private string GetRowValueForColumn(System.Data.DataRow row, string mappingColum
...
-            PolygonFeatureLayer impactAreaSet = new PolygonFeatureLayer("Impact_Area_Set", _impactAreaShapefile);
-
+            Projection terrainProjection = GetTerrainProjection(terrainLayer);
+            Projection siProjection = GetVectorProjection(pointShapefilePath);
+            Projection iaProjection = GetVectorProjection(ImpactAreaShapefilePath);
+            List<Polygon> impactAreas = LoadImpactAreasFromSourceFiles(ImpactAreaShapefilePath, terrainProjection);
+            PointFeatureLayer _structureFeatureLayer = new PointFeatureLayer("ThisNameIsNotUsed",pointShapefilePath);
             float[] groundelevs = Array.Empty<float>();
-            if (_updateGroundElevsFromTerrain)
+            int defaultMissingValue = -999;
```
[the remark this answers is not in the document]
Comment (Brennan1994) [reply fda-model/structures/Inventory.cs:122]: I'd be up for that if this is used anywhere else as our missing value. If it's only used here, which I think is the case, I think it's either cleaner, or comparable to leave it defined here. Do you know of another place we use a default integer value of -999?

[the remark this answers is not in the document]
Comment (rnugent3) [reply fda-model/structures/Inventory.cs:122]: looks like a lot of places and it is mostly my fault

image

[the remark this answers is not in the document]
Comment (rnugent3) [reply fda-model/structures/Inventory.cs:122]: I will put the fix in shortly

File: fda-model/structures/Inventory.cs [hunk 2]
```diff
@@ -158,112 +111,115 @@ private string GetRowValueForColumn(System.Data.DataRow row, string mappingColum
...
+            if (updateGroundElevFromTerrain)
             {
-                groundelevs = GetGroundElevationFromTerrain(_structureInventoryShapefile, _terrainPath);
+                groundelevs = GetGroundElevationFromTerrain(_structureFeatureLayer, terrainLayer);
             }
-
-            int defaultMissingValue = -999;
-            PointMs pointMs = new PointMs(structureInventory.Points().Select(p => p.PointM()));
-            for (int i = 0; i < structureInventory.FeatureCount(); i++)
+            foreach (PointM pt in pointMs)
```
[the remark this answers is not in the document]
Comment (Brennan1994) [reply fda-model/structures/Inventory.cs]: I think we should. That's now implemented.

**Comment by Brennan1994** [comment]
Philosophically I think it's unwise for me to push this change on the Friday before a class. Lets let this pull request sit. Work on the things that need to be done before class.

**Pull request merged**